### PR TITLE
Fix windows 10 test

### LINKED ISSUE #11577: Windows 10 reports directory modification times unreliably
When doing `Files.walkFileTree()` on Windows 10 the last modified timestamp of directories reported via the `BasicFileAttributes` parameter passed to `FileVisitor.preVisitDirectory()` is different than the actual modification time observed via `Files.getLastModifiedTime()` or `File.lastModified()`.

This makes `DirectorySnapshotterAsDirectoryWalkerTest` fail.

Since this only seems to happen with directories, and we don't use directory modification dates for snapshotting (or anything else AFAIK), this probably has no actual effect on the production code.

---
cc: @gradle/build-cache

### PR DESCRIPTION
**Pull request by lptr**
Fixes #11577

### CODE REVIEW

File: subprojects/file-collections/src/main/java/org/gradle/api/internal/file/AttributeBaseFileTreeElement.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+package org.gradle.api.internal.file;
+
+import org.gradle.api.file.RelativePath;
+import org.gradle.internal.file.Chmod;
+import org.gradle.internal.file.Stat;
+
+import java.io.File;
+import java.nio.file.attribute.BasicFileAttributes;
+
+public class AttributeBaseFileTreeElement extends DefaultFileTreeElement {
```
Comment (wolfs) [inline subprojects/file-collections/src/main/java/org/gradle/api/internal/file/AttributeBaseFileTreeElement.java]:
```suggestion
public class AttributeBasedFileTreeElement extends DefaultFileTreeElement {
```

File: subprojects/file-collections/src/main/java/org/gradle/api/internal/file/AttributeBaseFileTreeElement.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    private final BasicFileAttributes attributes;
+
+    public AttributeBaseFileTreeElement(File file, RelativePath relativePath, Chmod chmod, Stat stat, BasicFileAttributes attributes) {
```
Comment (wolfs) [inline subprojects/file-collections/src/main/java/org/gradle/api/internal/file/AttributeBaseFileTreeElement.java]: This class actually seems to be unused. Let's remove it!

File: subprojects/file-collections/src/main/java/org/gradle/api/internal/file/DefaultFileVisitDetails.java [hunk 3]
```diff
@@ -26,45 +26,16 @@
...
-        this.lastModified = lastModified;
-        this.size = size;
     }
 
     public DefaultFileVisitDetails(File file, Chmod chmod, Stat stat) {
         this(file, new RelativePath(!file.isDirectory(), file.getName()), new AtomicBoolean(), chmod, stat);
     }
 
-    @Override
-    public boolean isDirectory() {
```
Comment (wolfs) [inline subprojects/file-collections/src/main/java/org/gradle/api/internal/file/DefaultFileVisitDetails.java]: I think we should keep remembering if the file was a directory or not. Or use the information from `relativePath`.

File: subprojects/file-collections/src/main/java/org/gradle/api/internal/file/AttributeBaseFileTreeElement.java (hunk 1 above)
Comment (lptr) [reply subprojects/file-collections/src/main/java/org/gradle/api/internal/file/AttributeBaseFileTreeElement.java]: We don't actually use this class.

File: subprojects/file-collections/src/main/java/org/gradle/api/internal/file/AttributeBaseFileTreeElement.java (hunk 2 above)
Comment (lptr) [reply subprojects/file-collections/src/main/java/org/gradle/api/internal/file/AttributeBaseFileTreeElement.java]: Yeah.

File: subprojects/file-collections/src/main/java/org/gradle/api/internal/file/DefaultFileVisitDetails.java (hunk 3 above)
Comment (lptr) [reply subprojects/file-collections/src/main/java/org/gradle/api/internal/file/DefaultFileVisitDetails.java]: 👍 Fixing.

**Pull request merged**